src/airobot/utils/transform_util.py

```diff
@@ -0,0 +1,326 @@
+        position (Position): 3D position.
+        orientation (Orientation): 3D orientation (quaternion).
+    """
+    def __init__(self, position, orientation):
+        self.position = position
+        self.orientation = orientation
+
+
+class Header:
+    """
+    Class to represent the header that would accompany a
+    6D pose, containing a frame name.
+
+    Attributes:
+        frame_id (string): Name of coordinate frame.
+    """
+    def __init__(self):
+        self.frame_id = "world"
+
+
+class PoseStamped():
+    """
+    Class to represent a 6D pose, expressed in a particular frame.
+
+    Attributes:
+        pose (Pose): 6D pose.
+        header (Header): Header, with frame information.
+    """
+    def __init__(self):
+        position = Position()
+        orientation = Orientation()
+        pose = Pose(position, orientation)
+        header = Header()
+        self.pose = pose
+        self.header = header
+
+
+def pose_stamped2list(msg):
+    """
+    Function to convert a pose_stamped into a list
+
+    Args:
+        msg (PoseStamped): 6D pose.
+
+    Returns:
+        list: 6D pose, as list [x, y, z, qx, qy, qz, qw].
+    """
+    return [float(msg.pose.position.x),
+            float(msg.pose.position.y),
+            float(msg.pose.position.z),
+            float(msg.pose.orientation.x),
+            float(msg.pose.orientation.y),
+            float(msg.pose.orientation.z),
+            float(msg.pose.orientation.w),
+            ]
+
+def list2pose_stamped(pose, frame_id="world"):
+    """
+    Function to convert a list into a pose_stamped
+
+    Args:
+        pose (list): 6D pose, as list [x, y, z, qx, qy, qz, qw].
+        frame_id (str, optional): Frame this pose is expressed in.
+
+    Returns:
+        PoseStamped: 6D pose as a PoseStamped.
+    """
+    msg = PoseStamped()
+    msg.header.frame_id = frame_id
+    msg.pose.position.x = pose[0]
+    msg.pose.position.y = pose[1]
+    msg.pose.position.z = pose[2]
+    msg.pose.orientation.x = pose[3]
+    msg.pose.orientation.y = pose[4]
+    msg.pose.orientation.z = pose[5]
+    msg.pose.orientation.w = pose[6]
+    return msg
+
+
+def unit_pose():
+    """
+    Function to create a canonical pose, centered as the origin
+    and not rotated about any axis.
+
+    Returns:
+        PoseStamped: 6D canonical pose.
+    """
+    return list2pose_stamped([0, 0, 0, 0, 0, 0, 1])
+
+
+def matrix_from_pose(pose):
+    """
+    Function to convert from a PoseStamped to a 
+    homogeneous transformation matrix.
+
+    Args:
+        pose (PoseStamped): 6D pose, to convert to transformation matrix. 
+
+    Returns:
+        np.ndarray: Homogeneous transformation matrix representation of 
+            the pose.
+    """
+    pose_list = pose_stamped2list(pose)
+    trans, quat = pose_list[:3], pose_list[3:]
+    T = np.eye(4)
+    if hasattr(R, 'as_matrix'):
+        T[:-1, :-1] = R.from_quat(quat).as_matrix()
+    else:
+        T[:-1, :-1] = R.from_quat(quat).as_dcm()
```

Comment:
can we replace these lines with the existing functions in common.py

Reply:
fixed in 2259946